src/Goteo/Repository/DataSetRepository.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/*
+ * This file is part of the Goteo Package.
+ *
+ * (c) Platoniq y Fundación Goteo <fundacion@goteo.org>
+ *
+ * For the full copyright and license information, please view the README.md
+ * and LICENSE files that was distributed with
+ */
+
+ namespace Goteo\Repository;
+
+ use Goteo\Application\Exception\ModelException;
+ use Goteo\Application\Exception\ModelNotFoundException;
+ use Goteo\Model\DataSet;
+ use PDO;
+ use PDOException;
+
+ class DataSetRepository extends BaseRepository {
+     protected ?string $Table = 'data_set';
+
+     public function getById(int $id): DataSet  {
+         $sql = "SELECT data_set.*
+                FROM data_set
+                WHERE data_set.id = ?";
+
+         $dataSet = $this->query($sql, [$id])->fetchObject(DataSet::class);
+
+         if (!$dataSet instanceOf DataSet)
+             throw new ModelNotFoundException("DataSet with id $id not found");
+
+         return $dataSet;
+    }
+
+    public function getList(array $filter, int $offset = 0, int $limit = 10, bool $count = false) {
```

Comment:
The `$filter` property is not being used, so we better remove it.

Reply:
The $filter is not being used at the moment. I've defined it because it's how we used to create the getList functions before, as we could use this $filter array to include new conditions to the query, like searching by dates or whatever. We could change and then just create a getDataSetsBetweenDates or another function like that.

Reply:
I see what you mean, however, I still think that if is not needed right now, it should be removed. Adding it later is very simple and quick.